conda_forge_tick/migrators/arch.py

```diff
@@ -139,3 +139,132 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
         return super().remote_branch(feedstock_ctx) + "_arch"
+
+
+class OSXArm(GraphMigrator):
+    """
+    A Migrator that add arm osx builds to feedstocks
+    """
+
+    migrator_version = 1
+    rerender = True
+    # We purposefully don't want to bump build number for this migrator
+    bump_number = 0
+
+    ignored_packages = {"gfortran"}
+    excluded_dependencies = {"fortran_compiler_stub"}
+
+    arches = ["osx_arm64"]
+
+    additional_keys = {
+        "build_platform": {"osx_arm64": "osx_64"},
+        "test_on_native_only": True,
+    }
+
+    def __init__(
+        self, graph: nx.DiGraph = None, name: Optional[str] = None, pr_limit: int = 0,
+    ):
+        # rebuild the graph to only use edges from the arm osx requirements
+        graph2 = nx.create_empty_copy(graph)
+        for node, attrs in graph.nodes(data="payload"):
+            for plat_arch in self.arches:
+                deps = set().union(
+                    set(
+                        as_iterable(
+                            attrs.get(
+                                f"{plat_arch}_requirements",
+                                attrs.get("requirements", {}),
+                            ).get("host", set()),
+                        ),
+                    ),
+                )
+                for dep in deps:
+                    dep = graph.graph["outputs_lut"].get(dep, dep)
+                    graph2.add_edge(dep, node)
+
+        super().__init__(graph=graph2, pr_limit=pr_limit, check_solvable=False)
+
+        assert (
+            not self.check_solvable
+        ), "We don't want to check solvability for arm osx!"
+
+        # We are constraining the scope of this migrator
+        with indir("../conda-forge-pinning-feedstock/recipe/migrations"), open(
+            "osx_arm64.txt", "r",
+        ) as f:
+            self.target_packages = set(f.read().split())
+
+        self.name = name
+        # filter the graph down to the target packages
+        if self.target_packages:
+            self.target_packages.add("python")  # hack that is ~harmless?
+            packages = self.target_packages.copy()
+            for target in self.target_packages:
+                if target in self.graph.nodes:
+                    packages.update(nx.ancestors(self.graph, target))
+            self.graph.remove_nodes_from([n for n in self.graph if n not in packages])
+
+        # filter out stub packages and ignored packages
+        for node, attrs in list(self.graph.nodes("payload")):
+            if not attrs:
+                print(node)
+            if (
+                not attrs
+                or node.endswith("_stub")
+                or (node.startswith("m2-"))
+                or (node.startswith("m2w64-"))
+                or (node in self.ignored_packages)
+                or (attrs.get("meta_yaml", {}).get("build", {}).get("noarch"))
+            ):
+                pluck(self.graph, node)
+        for name in self.excluded_dependencies:
```

Comment:
We need this block before the above one since the above block removes the stubs we need.